zanata-war/src/main/java/org/zanata/service/impl/LocaleServiceImpl.java

```diff
@@ -90,12 +91,17 @@ public LocaleServiceImpl(LocaleDAO localeDAO, ProjectDAO projectDAO,
 
     public static Map<LocaleId, String> getLocaleAliasesByIteration(HProjectIteration iteration) {
         Map<LocaleId, String> localeAliases;
-        if (iteration.isOverrideLocales()) {
-            localeAliases = iteration.getLocaleAliases();
-        } else {
-            localeAliases = iteration.getProject().getLocaleAliases();
+        try {
+            if (iteration.isOverrideLocales()) {
+                localeAliases = iteration.getLocaleAliases();
+            } else {
+                localeAliases = iteration.getProject().getLocaleAliases();
+            }
+            return localeAliases;
+        }catch (NullPointerException e) {
+            e.printStackTrace();
```

Comment:
Catching `NullPointerException` is hardly ever the best option (except _maybe_ for a throwaway local code change). Catching but not propagating the exception is generally going to lead to more problems. Besides, Zanata should already be logging the stack trace elsewhere, as long as the exception is propagated up the call stack.

Also, if you do want to log an exception's stack trace, it's better to use a Logger, eg `log.error("unexpected error", e) instead of dumping to `stderr`.